### PR TITLE
Rargb: Update category mappings. Resolves #7760

### LINKED ISSUE #7760: [RARBG] - Category 54 not available/returned
# Environment
Docker (Ubuntu 18.04.4 LTS)

**OS**: Docker OS version: Unix 5.3.0.42 (64bit OS) (64bit process)

**.Net Runtime**: [.Net-Core/.Net-Framework/Mono]

**.Net Version**:

**Jackett Version**: Jackett v0.14.136.0

**Last Working Jackett Version**: N/A?

**Are you using a proxy or VPN?** [yes/no]
Yes

### Description

RARBG seems to have a category called “Movies/x265/1080” (category 54?) which is not able to be selected / returned in the jackett RARBG indexer?

### Logged Error Messages
N/A

### PR DESCRIPTION
**Pull request by cadatoiva**
missing Games/PS4 and Movies/x265/1080

remove duplicate category 49

sort categories by ID.

@garfield69, after digging into the functions when seeing the duplicated category, I found no reason why we can't duplicate the category if they contain more than one category of results. Was there a particular reason you can remember as to why we made it a practice to avoid this?

### CODE REVIEW

**Comment by garfield69** [comment]
It was my understanding that duplicate categories are not supported.
if you look at the config on an active RARBG indexer you will find only one entry, the first one defined in the addcategory list of the source code.
image
I presumed that any duplicates, while perhaps loaded into the category table, would never actually get referenced, as the routine would find the first match and ignore the rest.
Are you saying different?

**Comment by cadatoiva** [comment]
So, it's complicated. This is the code that adds a category to an indexer:
https://github.com/Jackett/Jackett/blob/7c33c536bc604e8c8740fbf1768bb0f093a0b3f2/src/Jackett.Common/Indexers/BaseIndexer.cs#L663-L688

In line 665 we always add a category mapping, which is what we use for `MapTorznabCapsToTracker()` giving us the categories to search in for a given standard category. Then we check if the Torznab category already exists in the tracker, so we can tell Radarr/Sonarr if the tracker supports movie/tv searches. In lines 666-671. Then we see if the tracker's category has a description and is a number. Then we add +100000 to the number and add it as a custom category if it's a unique number in lines 674 to the end.

So in summary, we always map the category to the Torznab standard, even for duplicates. But we'll skip adding a particular mapping as a custom category when there's no description, the category isn't a number, or the number is already used for a different custom category. The custom categories only really affect filtering on Jackett UI manual searching if I understand correctly.

**Comment by cadatoiva** [comment]
If I'm reading the code correctly, then like I asked about in #7664 if a category has movies and tv-shows, we can do something like:

```C#
AddCategoryMapping(42, TorznabCatType.MoviesBluRay, "Site Category Name");
AddCategoryMapping(42, TorznabCatType.TVUHD, Null);
```

And then when we do a Movie or TV search `MapTorznabCatsToTracker(query)` will return 42 for both. But when manual searching we can select category `100042 - Site Category Name` to get only the results from this category instead of all TV or Movie categories.

File: src/Jackett.Common/Indexers/Rarbg.cs [hunk 1]
```diff
@@ -80,28 +80,29 @@ public Rarbg(IIndexerConfigurationService configService, Utils.Clients.WebClient
...
             AddCategoryMapping(35, TorznabCatType.BooksEbook, "e-Books");
+            AddCategoryMapping(40, TorznabCatType.ConsolePS3, "Games/PS3");
+            AddCategoryMapping(41, TorznabCatType.TVHD, "TV HD Episodes");
+            AddCategoryMapping(42, TorznabCatType.MoviesBluRay, "Movies/Full BD");
+            AddCategoryMapping(44, TorznabCatType.MoviesHD, "Movies/x264/1080");
+            AddCategoryMapping(45, TorznabCatType.MoviesHD, "Movies/x264/720");
+            AddCategoryMapping(46, TorznabCatType.MoviesBluRay, "Movies/BD Remux");
+            AddCategoryMapping(47, TorznabCatType.Movies3D, "Movies/x264/3D");
+            AddCategoryMapping(48, TorznabCatType.MoviesHD, "Movies/XVID/720");
+            AddCategoryMapping(49, TorznabCatType.TVUHD, "TV UHD Episodes");        // torrentapi.org returns "Movies/TV-UHD-episodes" for some reason
```
Comment (garfield69) [inline src/Jackett.Common/Indexers/Rarbg.cs]: Um. I think we need to continue to keep both cat 49 versions in this case.
The first cat49 with description `TV UHD Episodes` is what the Web site displays when using a browser, and is what the user will be looking for in our dashboard filter.
However, the second cat49  with description `Movies/TV-UHD-episodes` is what the API actually returns in its json `"category":"Movies\/TV-UHD-episodes"`, and the code at 
https://github.com/Jackett/Jackett/blob/d5a540acaeb418a4e97d06a8048548640fcd2b68/src/Jackett.Common/Indexers/Rarbg.cs#L239
uses that description to do the matching to find and assign the correct cat number 49.

Comment (cadatoiva) [reply src/Jackett.Common/Indexers/Rarbg.cs]:
```suggestion
            AddCategoryMapping(49, TorznabCatType.TVUHD, "TV UHD Episodes");        // torrentapi.org returns "Movies/TV-UHD-episodes" for some reason
            AddCategoryMapping(49, TorznabCatType.TVHD, "Movies/TV-UHD-episodes"); // possibly because thats what the category is called on the /top100.php page
```

Comment (cadatoiva) [reply src/Jackett.Common/Indexers/Rarbg.cs]: Fixed. And that just further shows that duplicate categories are saved and used under the hood, just not displayed in the custom filters.
